src/main/scala/com/carta/excel/AppendOnlySheetWriter.scala

```diff
@@ -205,6 +209,10 @@ class AppendOnlySheetWriter
         substituteString(templateCell, outputCell, substitutionMap)
       case CellType.STRING =>
         outputCell.setCellValue(templateCell.getStringCellValue)
+      case CellType.FORMULA =>
+        val cellRefShiftFactor = numRowsCreated - templateCell.getRowIndex
+        val outputFormula = cellFormulaParser.shiftRowNums(templateCell.getCellFormula, cellRefShiftFactor)
```

Comment:
so we write the formula result in and not have excel just render it for us?

Reply:
wha, no, we adjust the cell references of the template formula to account for the repeated rows that we may have written above
excel is still responsible for calculating the formula from the cells in the output workbook

Reply:
what does `there cannot be repeated rows between any cell references.` mean then?

Reply:
so, if you have
```
cellA
<repeatedRow>
cellB
formula(cellA, cellB, cellC, cellD)
cellC
<repeatedRow>
cellD
```

using cellA and cellD is illegal in the formula cause there're repeated rows between the referenced cells and the formula itself.

if we track the row number of each repeated section and the number of rows created in the output sheet for each number, allowing cellA shouldn't be too bad

cellD is a bigger problem because of our append-only nature -- we can't go back to modify the formula we put in the output sheet, and we don't know how many rows there'll be when we copy the formula to the output sheet.

but, any static cells referenced between the two repeated sections are fine to be used in the template formula

Reply:
I see, so do we shift sum ranges too? ie. `=SUM(A1:A1)` and then if its a repeated row, do we shift A2 to be the last row?

Reply:
if you had
```
<repeated row>
cellA
cellB
cellC
cellD
formula(A:D)
```
Right now, the A:D reference would get shifted down to the correct rows of the output workbook based on the number of repeated rows we insert. From this PR, formulas that operate on repeated rows will not work.

That being said, this gave me an idea for how we could handle repeated rows with the append-only writer. For every repeated row in the template, we store a map of the template's row number to the start and end row number of the output sheet. Then for formula that reference a cell in a repeated row, we use that map to get the correct output range, and replace the reference with that range.

So the template could be
```
<repeated row with cell of interest at A4>
SUM(A4)
```
the user supplies, say, 6 rows to populate the repeated row
and then we could replace the `SUM(A4)` formula with `SUM(A4:A10)` in the output sheet